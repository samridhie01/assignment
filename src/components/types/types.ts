export interface PlanetData{
    name: string,
    climate: string
  }

export interface PlanetsProps {
    data: PlanetData;
    onSaveClick: (element: any) => void;
}
