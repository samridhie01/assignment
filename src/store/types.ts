import { addPlanetData } from '../actions/index';
import { ADD_PLANETDATA } from '../constants/action-types';

export interface Planet {

    climate: string;
    created: Date;
    diameter: number;
    edited: Date;
    films: string[];
    gravity: string;
    name: string;
    orbital_period: string;
    population: string;
    residents: string[];
    rotation_period: string;
    surface_water: string;
    terrain: string;
    url: string;
  }

  export interface PlanetState {
    planetData: Planet[];
  }

  interface AddPlanetData{
      type: typeof ADD_PLANETDATA,
      payload: Planet
      
  }

  export type PlanetActionTypes = AddPlanetData