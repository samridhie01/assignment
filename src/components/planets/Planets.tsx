import React, {useState, useEffect} from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Box, IconButton } from "@material-ui/core";
import PlanetTile from "./PlanetTile";
import FavoriteBorderIcon from "@material-ui/icons/FavoriteBorder";
import { PlanetsProps } from "../types/types";
import { Theme } from "@material-ui/core";


export interface StyleProps {
  color: string;
}

const useStyles = makeStyles<Theme, StyleProps>({
  flexContainer: {
    display: "flex",
    flexFlow: "row wrap",
    justifyContent: "space-around",
    float: "right"
  },
  btnColor: {
    color: props => props.color
  },
  endPos: {
    padding: "10px"
  }
});



const Planets: React.FC<PlanetsProps> = ({ data, onSaveClick }) => {
  const [myColor, setMyColor] = useState("black");
  const handleSave = () => {
    setMyColor("red");
    onSaveClick(data);
  };
  const classes = useStyles({ color: myColor });

  useEffect(() => {
    const addedFavs = localStorage.getItem("favs");
    if (addedFavs != null && addedFavs.includes(data.name)) {
      setMyColor("red");
    }
  }, [data.name]);

  return (
    <Box m={2} border={1} width={800} className={classes.flexContainer}>
      <PlanetTile name={data.name} climate={data.climate} />
      <Box m={4} border={1} width={200} height={150}>
        <div className={classes.endPos}>
          <p>If you like this planet, click heart Icon </p>
          <IconButton data-testId={"favIcon"} className={classes.btnColor} onClick={handleSave}>
            <FavoriteBorderIcon />
          </IconButton>
        </div>
      </Box>
    </Box>
  );
};

export default Planets;
