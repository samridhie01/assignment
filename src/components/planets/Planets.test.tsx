import {render, fireEvent, getByTestId} from "@testing-library/react";
import React from "react";
import Planets from "./Planets";
import {PlanetData} from "../types/types";
import {BrowserRouter} from "react-router-dom";

describe("Search Box", () => {
    const mockData: PlanetData = {
        name: "abc",
        climate: " def"
    }

    const renderComponent = render(
        <BrowserRouter>
            <Planets data={mockData} onSaveClick={jest.fn}/>
        </BrowserRouter>
    );

    it('should render snapshot', () => {
        const {asFragment} = renderComponent;
        expect(asFragment()).toMatchSnapshot();
    });

    it("should display fav icon against planet tile", ()=>{
        const {getByTestId} = renderComponent;
        const favBtn = getByTestId("favIcon")
        expect(favBtn).toBeInTheDocument();
    })




})
