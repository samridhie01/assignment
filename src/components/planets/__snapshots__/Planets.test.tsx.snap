// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-5 makeStyles-flexContainer-1"
  >
    <svg>
      planet.svg
    </svg>
    <div
      class="makeStyles-flexContainer2-6"
    >
      <h3>
        abc
      </h3>
      <div
        class="MuiBox-root MuiBox-root-8 makeStyles-item-7"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2"
        >
           def climate
        </h6>
      </div>
      <br />
      <a
        href="/details?search=abc"
      >
        View details
      </a>
    </div>
    <div
      class="MuiBox-root MuiBox-root-9"
    >
      <div
        class="makeStyles-endPos-3"
      >
        <p>
          If you like this planet, click heart Icon 
        </p>
        <button
          class="MuiButtonBase-root MuiIconButton-root makeStyles-btnColor-2 makeStyles-btnColor-4"
          data-testid="favIcon"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
